### PR TITLE
add set image command

### PR DESCRIPTION
**Pull request by narg95**
# What this PR does

This PR is the part 2 of [#402](https://github.com/kubernetes-sigs/kustomize/issues/402), with the following key features:

1. Adds the `kustomize edit set image` command:

```
$ kustomize edit set image -h
The command
  set image postgres=my-registry/postgres:latest nginx:1.8.0 my-app=my-registry/my-app alpine@sha256:24a0c4b4a4c0eb97a1aabb8e29f18e917d05abfe1b7a7c07857230879ce7d3d3
will add 

image:
- name: postgres
  newName: my-registry/postgres
  newTag: latest
- name: nginx
  newTag: 1.8.0
- name: my-app
  newName: my-registry/my-app
- digest: sha256:24a0c4b4a4c0eb97a1aabb8e29f18e917d05abfe1b7a7c07857230879ce7d3d3
  name: alpine

to the kustomization file if it doesn't exist,
and overwrite the previous ones if the image name exists.
```

2. Shows deprecation mesasge when using `set imagetag` command.

```
$ kustomize edit set imagetag -h

The `imagetag` command is deprecated, instead use `edit set image`.

Usage:
  kustomize edit set imagetag [flags]

Examples:
...
```

### CODE REVIEW

File: pkg/commands/edit/set/setimage.go [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+// newCmdSetImage sets the new names, tags or digests for images in the kustomization.
+func newCmdSetImage(fsys fs.FileSystem) *cobra.Command {
+	var o setImageOptions
+
+	cmd := &cobra.Command{
+		Use:   "image",
+		Short: `Sets images and their new names, new tags or digests in the kustomization file`,
+		Example: `
+The command
```
Comment (Liujingfang1) [inline pkg/commands/edit/set/setimage.go]: Can we separate this example? One is for `<image>=<newimage>:<newtag>` and `<image>=<newimage>@<newtag>`? The other is for `<image>:<newtag>` and `<image>@<digest>`.

File: pkg/commands/edit/set/setimage_test.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+				t.Errorf("unexpected read error: %v", err)
+				t.FailNow()
+			}
+			expectedStr := strings.Join(tc.expected.fileOutput, "\n")
+			if !strings.Contains(string(content), expectedStr) {
+				t.Errorf("unexpected image in kustomization file. \nActual:\n%s\nExpected:\n%s", content, expectedStr)
+			}
+		})
+	}
+}
```
Comment (monopole) [inline pkg/commands/edit/set/setimage_test.go]: Thanks for these tests.

File: pkg/commands/edit/set/setimagetag.go [hunk 3]
```diff
@@ -40,7 +41,7 @@ func newCmdSetImageTag(fsys fs.FileSystem) *cobra.Command {
 
 	cmd := &cobra.Command{
 		Use:   "imagetag",
-		Short: "Sets images and their new tags or digests in the kustomization file",
+		Short: "[*** DEPRECATED, use: kustomize edit set image ***] Sets images and their new tags or digests in the kustomization file",
```
Comment (monopole) [inline pkg/commands/edit/set/setimagetag.go]: Simply
```
The `imagetag` command is deprecated.  Instead use `edit set image`.
```

File: pkg/commands/edit/set/setimagetag.go [hunk 4]
```diff
@@ -58,6 +59,7 @@ to the kustomization file if it doesn't exist,
 and overwrite the previous ones if the image tag exists.
 `,
 		RunE: func(cmd *cobra.Command, args []string) error {
+			log.Print("This is a deprecated command, it still works but will be removed in a further release. Use better kustomize edit set image")
```
Comment (monopole) [inline pkg/commands/edit/set/setimagetag.go]:
```
log.Println(cmd.Short)
```

Comment (narg95) [reply pkg/commands/edit/set/setimagetag.go]: done

File: pkg/commands/edit/set/setimagetag.go (hunk 3 above)
Comment (narg95) [reply pkg/commands/edit/set/setimagetag.go]: 👍

File: pkg/commands/edit/set/setimage_test.go (hunk 2 above)
Comment (narg95) [reply pkg/commands/edit/set/setimage_test.go]: with pleasure :) !

File: pkg/commands/edit/set/setimage.go (hunk 1 above)
Comment (narg95) [reply pkg/commands/edit/set/setimage.go]: sounds good,  I also added  the case for `<image>=<newimage>` which is also supprted

**Pull request merged**